### DISCUSSION TITLE
GraphQL API - bulk create entities

### DISCUSSION THREAD

**Discussion by vanhock** [payloadcms/payload#1383] [Q&A]
# Bug Report
Hi there! 
I tried to extend GraphQL API and create an endpoint that allows generating entities from arrays.

Here is the code:
```
const productInputType = (GraphQL, fields) => {
  return new GraphQL.GraphQLInputObjectType({
    name: 'ProductInput',
    fields
  })
}

export default buildConfig({
...
graphQL: {
    mutations: (GraphQL, payload) => {
      const createProductScheme = payload.Mutation.fields.createProduct;
      return {
        createProducts: {
          type: new GraphQL.GraphQLList(createProductScheme.type),
          args: {
            productsArray: { type: new GraphQL.GraphQLList(productInputType(GraphQL, createProductScheme.args)) }
          }
        }
      }
    }
  }
```

## Current Behavior

It does not work :(

## Expected Behavior

Create multiple entities at one request

## Possible Solution

-

## Steps to Reproduce
-
## Detailed Description
-

**Comment by jmikrut** [comment]
Hey @vanhock — this should be totally possible. I'm going to convert this to a discussion because I believe it involves your specific implementation, but we're happy to help you figure this out. 

PS - bulk creation / edit / deletion is on our radar for building it into Payload itself. We're very likely to tackle this as our next big feature.

https://github.com/payloadcms/payload/discussions/1382

I can bump it up in priority for you!
